s-token/src/lib.rs

```diff
@@ -522,42 +511,13 @@ impl SToken {
         write_allowance(e, from, spender, allowance - amount);
     }
 
-    fn receive_balance(e: &Env, addr: Address, amount: i128) {
-        let balance = read_balance(e, addr.clone());
-        if !is_authorized(e, addr.clone()) {
-            panic!("can't receive when deauthorized");
-        }
-        write_balance(e, addr, balance + amount);
-    }
-
-    fn spend_balance(e: &Env, addr: Address, amount: i128) {
-        let balance = read_balance(e, addr.clone());
-        if !is_authorized(e, addr.clone()) {
-            panic!("can't spend when deauthorized");
-        }
-        if balance < amount {
-            panic!("insufficient balance");
-        }
-        write_balance(e, addr, balance - amount);
-    }
-
-    fn add_total_supply(e: &Env, amount: i128) {
-        let mut total_supply: i128 = read_total_supply(e);
-        total_supply = total_supply.checked_add(amount).unwrap();
-        if total_supply < 0 {
-            panic!("negative total supply");
-        }
-
-        write_total_supply(e, total_supply);
-    }
-
     fn do_mint(e: &Env, user: Address, amount: i128) {
         if amount == 0 {
             panic!("s-token: invalid mint amount");
```

Comment:
Sometimes we add a context in panic statements. Let's find a consistent way for it.